### PR TITLE
gyro calibration fix and add mavlink time offset

### PR DESCRIPTION
**Pull request by ChristophTobler**
I fixed the gyro calibration and added a time offset for the optical flow message so that it is in sync with the imu messages from the autopilot and we don't need to change the Firmware/estimator.

### CODE REVIEW

**Comment by zehortigoza** [comment]
Oh cool, thanks.

File: src/mainloop.cpp [hunk 1]
```diff
@@ -274,6 +280,17 @@ void Mainloop::camera_callback(const void *img, UNUSED size_t len, const struct
...
 
+static void _highres_imu_msg_callback(const mavlink_highres_imu_t *msg, void *data)
+{
+	Mainloop *mainloop = (Mainloop *)data;
+	mainloop->set_mavlink_time(msg);
+}
+
+void Mainloop::set_mavlink_time(const mavlink_highres_imu_t *msg)
+{
+	_mavlink_time_usec = msg->time_usec;
```
Comment (zehortigoza) [inline src/mainloop.cpp]: Maybe just set this once, so the time_usec will increment constantly

Comment (ChristophTobler) [reply src/mainloop.cpp]: I tried that but if I only set it once the offset is not as accurate

Comment (ChristophTobler) [reply src/mainloop.cpp]: However, I could use the bool if the offset has already been set

**Comment by ChristophTobler** [comment]
@zehortigoza I changed it so it only gets updated until the offset is set

**Comment by zehortigoza** [comment]
Merged with just a small change fixing the build, thanks

**Pull request closed without merging**